cmd/client-s3.go

```diff
@@ -1615,16 +1611,11 @@ func (c *S3Client) Stat(ctx context.Context, opts StatOptions) (*ClientContent,
 		if err == nil {
 			return ctnt, nil
 		}
-
 		// Ignore object missing error but return for other errors
 		if !errors.As(err.ToGoError(), &ObjectMissing{}) && !errors.As(err.ToGoError(), &ObjectIsDeleteMarker{}) {
 			return nil, err
 		}
-	}
-
-	// No object found, start looking for a prefix with the same name
-	// or a directory marker. Add a trailing slash if it is not in the path
-	if !strings.HasSuffix(path, string(c.targetURL.Separator)) {
+		// The object is not found, look for a directory marker or a prefix
```

Comment:
This is the only code change. (removing the `if`). Can you explain how that does what your description says?

I don't really understand.